src/hotspot/share/opto/phaseX.cpp

```diff
@@ -85,15 +110,12 @@ Node *NodeHash::hash_find( const Node *n ) {
   while( 1 ) {                  // While probing hash table
     if( k->req() == req &&      // Same count of inputs
         k->Opcode() == op ) {   // Same Opcode
-      for( uint i=0; i<req; i++ )
-        if( n->in(i)!=k->in(i)) // Different inputs?
-          goto collision;       // "goto" is a speed hack...
-      if( n->cmp(*k) ) {        // Check for any special bits
+      bool collision = check_for_collision(n, k);
+      if (collision == false && n->cmp(*k)) {  // Check for any special bits
```

Comment:
```suggestion
      if (!collision && n->cmp(*k)) {  // Check for any special bits
```

Reply:
A renaming of `check_for_collision` -> `have_equivalent_inputs` (with swap of true <-> false) would make this a bit more readable now, right?
```suggestion
      if (have_equivalent_inputs(n, k) && n->cmp(*k)) {  // Check for any special bits
```